packages/generate/src/syntax/toEdgeQL.ts

```diff
@@ -385,14 +391,16 @@ function walkExprTree(
     // const arg = (expr as any)?.__parent__ || (expr as any)?.__name__;
     // if (arg) console.log(arg);
     // }
-    seenExpr.parentScopes.add(parentScope);
+    if (parentScope) {
+      seenExpr.parentScopes.add(parentScope);
+    }
```

Comment:
Just a question for my own context: Was there some reason we were inserting `null` scopes before?

Reply:
Ah, you're right, there was a reason. We do need to track if an expr is used outside of a scope, so we can throw an error if it needs extracting to a `with` var, since there'll be no place to put it.

Reply:
Fixed